internal/server/repository_workerauth.go

```diff
@@ -250,6 +246,43 @@ func StoreNodeInformationTx(ctx context.Context, reader db.Reader, writer db.Wri
 		return new(types.DuplicateRecordError)
 	}
 
+	// It's possible a connection dropped during a rotate credentials response, so the control plane's stored
+	// previous and current WorkerAuth records may not match what the worker has stored.
+	// Check what the worker indicates is its previous key and fix what we have stored before inserting the new record.
+	if node.PreviousEncryptionKey != nil {
```

Comment:
We were running into this when the response from cred rotation from the controller is dropped- the controller stores what it thinks is the correct previous/ next pair for the worker, while the worker discards the current cred it proposed since it failed to get a response. There's no way for the worker to know whether the controller got its request or only dropped the response, so we can't fix the issue there. We can only detect it when the worker sends another rotation request (which it will, since it assumes the first one failed), so that's what we're doing here.